src/Command/ProcessCommand.php

```diff
@@ -57,8 +58,17 @@ final class ProcessCommand extends Command
 
     private int $maximumNumberOfTries;
 
+    /** @var ImageResource[] */
+    private array $processableImageResources = [];
+
+    private ?int $limitPerResource = null;
```

Comment:
```suggestion
    private int $limitPerResource = 0;
```

Often times `0` means no limit and maybe it would simplify some things?

Reply:
If I add limit=0 as input I except the limit to be 0. Not infinite.

Reply:
Then I would argue we need an if statement:

```php
if(0 === $limit) {
    echo "You don't get it, newb";
}
```
Jokes aside - I don't quite get your argument here. You would expect the command to not run when you give it an input that is allowed?

Reply:
Also I don't have a problem with the nullability. I just thought it could simplify some things. That is not given of course

Reply:
I mean. We have "reasonable" type system that allows you to encode optionality (nullable). So why would I not use that?

If someone goes out of the their way to specify `--limit=0`, do you think they just wanted the default value? :)

In another scenario. Imagine you have another process that calls this command, and they calculate limit based on *something*, and it calculated that 0 more was needed (and for some reason still calls the command), it would be quite the surprise if it processes everything.

Reply:
You should have a PhD i edge cases 😂

Reply:
But yes, let's keep the nullability, but not allow a limit of 0, which makes no sense. I think it's fine to tell people when things doesn't make sense